cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/MetadataSubscriberService.java

```diff
@@ -180,6 +190,60 @@ protected boolean shouldRunInSeparateTx(ImmutablePair<String, MetadataMessage> m
     return entityType.equals(EntityType.INSTANCE) || entityType.equals(EntityType.NAMESPACE);
   }
 
+  @Override
+  protected void doStartUp() throws Exception {
+    LOG.debug("Starting scheduled service {}", getServiceName());
+
+    LOG.debug("Starting backfill process for plugin metadata");
+
+    // Backfill plugin metadata
+    NamespaceStore namespaceStore = new DefaultNamespaceStore(this.transactionRunner);
```

Comment:
shouldn't do work like this in a start up method. Services should start up quickly and perform work after they're started.

For example, you could potentially override the runTask() method to check if backfill is done. If done, call super.runTask(). If not, backfill, set a flag, then call super.runTask().

Reply:
I see this class is using the StoreDefinition.AppMetadataStore.SUBSCRIBER_MESSAGE table to keep track of the subscriber state. You could do something similar to track whether or not backfill was already done, so that this doesn't happen every time during startup. Could save this for a follow up PR.

Reply:
Seems like `runTask()` was defined as final in the parent class. I created a new method in the parent class `preProcess()` which runs before messages are processed and moved the backfill code into there.

I also added a max retry check incase backfill fails more than 10 times.